scraper/ordoclic.py

```diff
@@ -0,0 +1,111 @@
+import httpx
+from httpx import TimeoutException
+import json
+from datetime import datetime, timedelta
+
+timeout = httpx.Timeout(15.0, connect=15.0)
+session = httpx.Client(timeout=timeout)
+
+# get all slugs
+def search():
+    base_url = 'https://api.ordoclic.fr/v1/public/search'
+    #payload = {'page': '1', 'per_page': '10000', 'in.isCovidVaccineSupported': 'true', 'in.isPublicProfile': 'true' }
+    payload = {'page': '1', 'per_page': '10000', 'in.isPublicProfile': 'true' }
+    r = session.get(base_url, params=payload)
+    r.raise_for_status()
+    return(r.json())
+  
+def getReasons(entityId):
+    base_url = f'https://api.ordoclic.fr/v1/solar/entities/{entityId}/reasons'
+    r = session.get(base_url)
+    r.raise_for_status()
+    return(r.json())
+
+def getSlots(entityId, medicalStaffId, reasonId, start_date, end_date):
+    base_url = 'https://api.ordoclic.fr/v1/solar/slots/availableSlots'
+    payload = {"entityId": entityId, 
+               "medicalStaffId": medicalStaffId, 
+               "reasonId": reasonId,
+               "dateEnd": f"{end_date}T00:00:00.000Z", 
+               "dateStart": f"{start_date}T23:59:59.000Z"}
+    headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+    r = session.post(base_url, data=json.dumps(payload), headers=headers)
+    r.raise_for_status()
+    return(r.json())
+
+def getProfile(rdv_site_web):
+    profile_name = rdv_site_web.rsplit('/', 1)[-1]
+    base_url = f'https://api.ordoclic.fr/v1/public/entities/profile/{profile_name}'
+    r = session.get(base_url)
+    r.raise_for_status()
+    return(r.json())
+
+def parse_ordoclic_slots(availability_data):
+    first_availability = None
+    if not availability_data:
+        return None
+    if 'nextAvailableSlotDate' in availability_data:
+        nextAvailableSlotDate = availability_data.get('nextAvailableSlotDate', None)
+        if nextAvailableSlotDate != None:
+            first_availability = datetime.strptime(nextAvailableSlotDate, '%Y-%m-%dT%H:%M:%S%z') + timedelta(minutes=120)
```

Comment:
Avec `timedelta(minutes=120)`, on hardcode un décalage sur l'heure d'été de France métropolitaine, je ne recommande pas. A défaut de `timeStartUtcOffset` ou équivalent dans les données, je propose de gérer a minima l'heure métropolitaine d'hiver et d'été :
```suggestion
            first_availability = datetime.strptime(nextAvailableSlotDate, '%Y-%m-%dT%H:%M:%S%z')
            first_availability += first_availability.replace(tzinfo=timezone('CET')).utcoffset()
```
En oubliant pas `from pytz import timezone`